src/utils/zcl_abapgit_environment.clas.abap

```diff
@@ -238,4 +238,16 @@ CLASS zcl_abapgit_environment IMPLEMENTATION.
 
   ENDMETHOD.
 
+  METHOD zif_abapgit_environment~is_running_in_test_context.
+    " Note this approach only works for the developer version. The standalone version will always report not running in
+    " test context which should be fine as there are no unit tests delivered in it.
+    cl_abap_typedescr=>describe_by_name(
+      EXPORTING
+        p_name         = |\\PROGRAM={ sy-repid }\\CLASS=LTCL_TEST|
```

Comment:
How does that work? Isn't `SY-REPID` always `ZCL_ABAPGIT_ENVIRONMENT=======CP`? What if the class isn't called `LTCL_TEST`?

Reply:
`LTCL_TEST` is a test class with the `FOR TESTING ADDITION`. RTTI cannot access test classes unless it is called in a running unit test ;)

Reply:
I had a different understanding of how it should work. Got it now. 
It didn't work for me because I also mocked `ZIF_ABAPGIT_ENVIRONMENT~IS_RUNNING_IN_TEST_CONTEXT` with an empty implementation returning FALSE. Not sure whether that is an issue or not.

Reply:
Ah, interesting. Might as well move it out of the interface and directly call the method in the exit factory method or even move it to the exit class. I cannot think of another place where this check would make sense so it doesn't need to be that reusable.

Reply:
+1 for private method